### PR TITLE
Cleans up WalletStorage / UserType

### PR DESCRIPTION
**Pull request by Moejoe90**
There is still a second stage for this to be done after auth sagas refactor
https://github.com/Neufund/platform-frontend/issues/3246

### CODE REVIEW

File: app/components/modals/accept-tos-modal/AcceptTosModal.tsx [hunk 1]
```diff
@@ -78,11 +74,7 @@ export const AcceptTosModalInner: React.ComponentType<IStateProps & IDispatchPro
       </Button>
     </div>
     <div>
-      <Button
-        className={styles.rejectTos}
-        onClick={() => onLogout(userType)}
-        layout={EButtonLayout.INLINE}
-      >
+      <Button className={styles.rejectTos} onClick={() => onLogout()} layout={EButtonLayout.INLINE}>
```
Comment (desfero) [inline app/components/modals/accept-tos-modal/AcceptTosModal.tsx]: nitpick, `onClick={onLogout}`

File: app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx [hunk 2]
```diff
@@ -49,10 +42,7 @@ export const RecoverySuccessComponent: React.FunctionComponent<IStateProps & IDi
     </Col>
     <Row className="justify-content-center mb-5 mt-5 pt-4">
       <Col xs={6} sm={5} md={4} lg={3}>
-        <Button
-          onClick={() => goToDashboard(userType)}
-          data-test-id="recovery-success-btn-go-dashboard"
-        >
+        <Button onClick={() => goToDashboard()} data-test-id="recovery-success-btn-go-dashboard">
```
Comment (desfero) [inline app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx]: nitpic `onClick={goToDashboard}`

File: app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx [hunk 3]
```diff
@@ -69,10 +59,7 @@ export const RecoverySuccessComponent: React.FunctionComponent<IStateProps & IDi
 );
 
 export const RecoverySuccess = compose<React.FunctionComponent>(
-  appConnect<IStateProps, IDispatchProps>({
-    stateToProps: s => ({
-      userType: selectUrlUserType(s.router),
-    }),
+  appConnect<void, IDispatchProps>({
```
Comment (desfero) [inline app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx]: why just not `{}` for consistency over `void`?

File: app/modules/kyc/sagas.ts [hunk 4]
```diff
@@ -656,10 +655,7 @@ function* waitForKycStatusLoad(): Iterator<any> {
 
 export function* loadKycRequestData(): any {
   // Wait for contracts to init
-  const isSmartContractsInitialized = yield select(selectIsSmartContractInitDone);
-  if (!isSmartContractsInitialized) {
-    yield neuTakeOnly(actions.init.done, { initType: EInitType.START_CONTRACTS_INIT });
-  }
+  yield waitUntilSmartContractsAreInitialized();
```
Comment (desfero) [inline app/modules/kyc/sagas.ts]: 👍

File: app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx (hunk 3 above)
Comment (Moejoe90) [reply app/components/wallet-selector/wallet-recover/success/RecoverySuccess.tsx]: I was thinking if you put `{}` then you can still add some props and forget to change the type from lets say `{}` to `IStateProps` but the void will force you to change the type

**Pull request merged**